### ISSUE TITLE
Targeting Swift 4 in Xcode 10

### ISSUE THREAD

**Issue by seanrucker** [RoverPlatform/rover-ios#319]
When using Xcode 10 but targeting Swift 4, the compiler complains about the authorization status being exhaustive. I believe this is a iOS 11 vs 12 issue and so an additional conditional compilation step may be required.

![Screen Shot 2018-10-17 at 4.01.28 PM.png](https://images.zenhubusercontent.com/5bbcadbaa9a60b5456b2c98a/e772da73-8ecb-4e86-8a8a-507fd4bfe506)

**Comment by seanrucker** [comment]
In theory, the following _should_ work however, when compiling in Xcode 10 we still get the "not exhaustive" error.

```swift
if #available(iOS 12.0, *) {
    switch authorizationStatus {
    case .authorized:
        return "authorized"
    case .denied:
        return "denied"
    case .notDetermined:
        return "notDetermined"
    case .provisional:
        return "provisional"
    }
} else {
    switch authorizationStatus {
    case .authorized:
        return "authorized"
    case .denied:
        return "denied"
    default:
        return "notDetermined"
    }
}
```

Here's why it fails. The SDK level is _raised_ within the `#available` block allowing code within it to make API calls only available in iOS 12 and greater. The reverse is not true. The `else` block does not decrease the SDK level and therefore depending on which version of Xcode you are running the switch may no not be exhaustive, regardless of which Swift version you are targeting!

From what I can tall there is no way to decrease the SDK level. However since the else statement will only run on devices less than iOS 12 we can bypass the situation by using a `default` case and be sure the only way to hit the `default` is in the case of `notDetermined`. 

```swift
if #available(iOS 12.0, *) {
    switch authorizationStatus {
    case .authorized:
        return "authorized"
    case .denied:
        return "denied"
    case .notDetermined:
        return "notDetermined"
    case .provisional:
        return "provisional"
    }
} else {
    switch authorizationStatus {
    case .authorized:
        return "authorized"
    case .denied:
        return "denied"
    case .notDetermined:
        return "notDetermined"
    }
}
```

**Comment by seanrucker** [comment]
The above strategy still doesn't compile in Xcode 9 as the compiler complains about the provisional case. It appears there is no way (AFAIK) to capture the provisional case when the developer is using Xcode 10 but targeting a Swift version < 4.2. I blame this on Swift's conditional compilation. Swift version != SDK version!!!
